auth-starter/src/main/java/com/vaadin/auth/starter/VaadinAuthSecurityConfiguration.java

```diff
@@ -155,12 +194,12 @@ static class DefaultVaadinAuthContext implements VaadinAuthContext {
         private CompositeLogoutHandler logoutHandler;
 
         @Override
-        public Optional<OidcUser> getAuthenticatedUser() {
+        public Optional<VaadinUser> getAuthenticatedUser() {
```

Comment:
Should there be an advanced method with a signature like this: 

`public <T extends VaadinUser> Optional<T> getAuthenticatedUser(Class<T> userClass) `